plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/IssuesPublisher.java

```diff
@@ -243,4 +277,64 @@ private QualityGateEvaluationMode determineQualityGateEvaluationMode(final Repor
         }
         return qualityGateEvaluationMode;
     }
+
+    private String extractChecksSummary(final IssuesStatistics statistics) {
+        return String.format("## %d issues in total:\n"
+                        + "- ### %d new issues\n"
+                        + "- ### %d Outstanding Issues\n"
+                        + "- ### %d delta issues\n"
+                        + "- ### %d fixed issues",
+                statistics.getTotalSize(), statistics.getNewSize(), statistics.getTotalSize() - statistics.getNewSize(),
+                statistics.getDeltaSize(), statistics.getFixedSize());
+    }
+
+    private String extractChecksText(final IssuesStatistics statistics) {
+        return "## Total Issue Statistics:\n"
+                + generateSeverityText(statistics.getTotalLowSize(), statistics.getTotalNormalSize(),
+                statistics.getTotalHighSize(), statistics.getTotalErrorSize())
+                + "## New Issue Statistics:\n"
+                + generateSeverityText(statistics.getNewLowSize(), statistics.getNewNormalSize(),
+                statistics.getNewHighSize(), statistics.getNewErrorSize())
+                + "## Delta Issue Statistics:\n"
+                + generateSeverityText(statistics.getDeltaLowSize(), statistics.getDeltaNormalSize(),
+                statistics.getDeltaHighSize(), statistics.getDeltaErrorSize());
+    }
+
+    private String generateSeverityText(final int low, final int normal, final int high, final int error) {
+        return "* low: " + low + "\n"
+                + "* Normal: " + normal + "\n"
+                + "* High: " + high + "\n"
+                + "* Error: " + error + "\n";
+    }
+
+    private ChecksConclusion extractChecksConclusion(final QualityGateStatus status) {
+        switch (status) {
+            case INACTIVE:
+                return ChecksConclusion.SUCCESS;
+            case FAILED:
+                return ChecksConclusion.FAILURE;
+            case WARNING:
+                return ChecksConclusion.NEUTRAL;
+            case PASSED:
+                return ChecksConclusion.SUCCESS;
+            default:
+                throw new IllegalArgumentException("Unsupported quality gate status: " + status);
+        }
+    }
+
+    private List<ChecksAnnotation> extractChecksAnnotations(final Report issues) {
+        return issues.stream()
+                .map(issue -> new ChecksAnnotationBuilder()
+                        .withPath(issue.getFileName())
+                        .withTitle(issue.getType())
+                        .withAnnotationLevel(ChecksAnnotationLevel.WARNING)
+                        .withMessage(issue.getSeverity() + ": " + issue.getMessage())
+                        .withStartLine(issue.getLineStart())
```

Comment:
The warnings plugin uses the magic value of `0` to indicate a warning for the whole file. Maybe I should map that to real values in `analysis-model`. However, columns can be 0 if not used, is this a problem or do we need to suppress the call in such cases?

Reply:
GitHub accepts 0 for the line field, but when using 0, the annotation will not be shown on the file changes view, it will only show in the check run page, so maybe GitHub thinks the same way as you do. 

For columns, I didn't see anything different with or without it, same for multiple lines annotation :shrug:

Reply:
maybe use line 1?

Reply:
I have not met annotation for the whole file, what kind of these annotations are? Shouldn't the biggest unit in a file for this static analysis is class instead of the whole file?

Reply:
A search for `.hasLineStart(0)` in my test suite returns 77 matches 😄 So there are a lot of them. I don't know if that is useful though...